### PR TITLE
Nbansal/feedbackchanges

### PR DESCRIPTION
**Pull request by nbansal1993**
- [ ] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

### CODE REVIEW

File: packages/selenium-ide/src/background/background.js [hunk 1]
```diff
@@ -230,7 +230,10 @@ browser.runtime.onMessageExternal.addListener(
     let payload = message.payload
 
     payload.sender = sender.id
-    if (message.uri === '/privateConnect' || message.uri === '/privateClose') {
+    if (
+      message.uri === '/private/connect' ||
```
Comment (melode11) [inline packages/selenium-ide/src/background/background.js]: you can use something like uri start with /private/ in this condition.

File: packages/selenium-ide/src/neo/IO/filesystem.js [hunk 2]
```diff
@@ -72,7 +72,18 @@ export function loadAsText(blob) {
 
 export function saveProject(_project) {
   const project = _project.toJS()
-  downloadProject(project)
+  if (UiState.isControlled) {
+    const saveMessage = {
```
Comment (melode11) [inline packages/selenium-ide/src/neo/IO/filesystem.js]: That is not what I mean, can we move this to original place in 
 ? Promise.resolve()
And please make this send event a separated method.

**Review by melode11 — CHANGES_REQUESTED** [review]

File: packages/selenium-ide/src/neo/IO/filesystem.js (hunk 2 above)
Comment (nbansal1993) [reply packages/selenium-ide/src/neo/IO/filesystem.js]: If we move to original place, then we get 2 requests on save: one from plugin emit and one from this message. Tomer said plugin emit should not happen in control mode

File: packages/selenium-ide/src/background/background.js (hunk 1 above)
Comment (nbansal1993) [reply packages/selenium-ide/src/background/background.js]: done

Comment (melode11) [reply packages/selenium-ide/src/background/background.js]: Where is the change?

File: packages/selenium-ide/src/neo/IO/filesystem.js [hunk 3]
```diff
@@ -74,21 +74,25 @@ export function saveProject(_project) {
...
-    browser.runtime.sendMessage(Manager.controller.id, saveMessage)
+    sendProject(project)
   } else {
     //In normal mode download the project
     downloadProject(project)
   }
   UiState.saved()
 }
 
+function sendProject(project) {
```
Comment (melode11) [inline packages/selenium-ide/src/neo/IO/filesystem.js]: better to be sendSaveProjectEvent()

**Review by melode11 — APPROVED** [review]

**Pull request merged**